crates/turbopack-dev-server/src/source/combined.rs

```diff
@@ -1,21 +1,42 @@
+use std::mem;
+
 use anyhow::Result;
 use turbo_tasks::{primitives::StringVc, TryJoinIterExt, Value};
 use turbopack_core::introspect::{Introspectable, IntrospectableChildrenVc, IntrospectableVc};
 
 use super::{
-    specificity::SpecificityReadRef, ContentSource, ContentSourceData, ContentSourceResultVc,
-    ContentSourceVc,
+    specificity::SpecificityReadRef, ContentSource, ContentSourceContent, ContentSourceData,
+    ContentSourceResultVc, ContentSourceVc, NeededData,
 };
 use crate::source::ContentSourcesVc;
 
 /// Combines multiple [ContentSource]s by trying all content sources in order.
-/// First [ContentSource] that responds with something other than NotFound will
-/// serve the request.
+/// The content source which responds with the most specific response (that is
+/// not a [ContentSourceContent::NotFound]) will be returned.
 #[turbo_tasks::value(shared)]
 pub struct CombinedContentSource {
     pub sources: Vec<ContentSourceVc>,
 }
 
+/// A helper source which allows the [CombinedContentSource] to be paused while
+/// we ask for vary data.
+#[turbo_tasks::value(shared)]
+pub struct PausableCombinedContentSource {
+    /// The index of the item which requested vary data. When running [get], we
+    /// will skip to exactly this item to resume iteration.
+    index: usize,
+
+    /// The paused state (partially processed path, content source, vary data)
+    /// of the internal content source which asked for vary data.
+    pending: Option<NeededData>,
```

Comment:
Nit: Since we don't actually use `pending.vary`, this should be its own `PausableCombinedContentSourcePendingState` struct with only a `path` and a `source`.

That way, you would not need the fiddling below.

Reply:
Done.